src/main/kotlin/vm/Executable.kt

```diff
@@ -0,0 +1,35 @@
+package com.dlfsystems.vm
+
+import com.dlfsystems.value.VFun
+import com.dlfsystems.value.VObj
+import com.dlfsystems.value.Value
+import kotlinx.serialization.Serializable
+
+interface Executable {
+
+    @Serializable
+    data class Block(val start: Int, val end: Int)
+
+    val code: List<VMWord>
```

Comment:
it would be nice if these were constructor parameters, but Executable can't be an abstract class since VFun has to also inherit Value.   I could make a composition-style member for VFun to be the Executable but ehh.